RevenueCatUI/Templates/MultiPackageTemplate.swift

```diff
@@ -0,0 +1,241 @@
+
+            IntroEligibilityStateView(
+                textWithNoIntroOffer: package.localization.offerDetails,
+                textWithIntroOffer: package.localization.offerDetailsWithIntroOffer,
+                introEligibility: self.introEligibility[package.content],
+                foregroundColor: selected
+                    ? .white
+                    : .black,
+                alignment: alignment
+            )
+            .fixedSize(horizontal: false, vertical: true)
+            .font(.body)
+        }
+        .font(.body.weight(.medium))
+        .padding()
+        .multilineTextAlignment(.leading)
+        .frame(maxWidth: .infinity, alignment: alignment)
+        .foregroundColor(self.configuration.colors.foregroundColor)
+        .background {
+            RoundedRectangle(cornerRadius: 15, style: .continuous)
+                .foregroundColor(selected ? .black : .init(white: 0.8))
+        }
+    }
+
+    private var subscribeButton: some View {
+        PurchaseButton(
+            package: self.selectedPackage,
+            purchaseHandler: self.purchaseHandler,
+            colors: self.configuration.colors,
+            localization: self.selectedLocalization,
+            introEligibility: self.introEligibility[self.selectedPackage],
+            mode: self.configuration.mode
+        )
+    }
+
+    @ViewBuilder
+    private var backgroundImage: some View {
+        if let url = self.configuration.backgroundURL {
+            RemoteImage(url: url)
+                .blur(radius: 40)
+                .opacity(0.7)
+        } else {
+            DebugErrorView("Template configuration is missing background URL",
+                           releaseBehavior: .emptyView)
+        }
+    }
+
+    @ViewBuilder
+    private var iconImage: some View {
+        if let url = self.configuration.iconURL {
+            RemoteImage(url: url, aspectRatio: 1)
+                .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))
+                .frame(maxWidth: 100)
+        } else {
+            DebugErrorView("Template configuration is missing icon URL",
+                           releaseBehavior: .emptyView)
+        }
+    }
+
+}
+
+// MARK: - Extensions
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+private extension MultiPackageTemplateContent {
+
+    func localization(for package: Package) -> ProcessedLocalizedConfiguration {
+        // Because of how packages are constructed this is known to exist
+        return self.localization[package]!
+    }
+
+    var selectedLocalization: ProcessedLocalizedConfiguration {
+        return self.localization(for: self.selectedPackage)
+    }
+
+}
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+private extension TemplateViewConfiguration {
+
+    var backgroundURL: URL? {
+        guard self.imageURLs.count >= 2 else { return nil }
+        return self.imageURLs[1]
+    }
+
+    var iconURL: URL? {
+        return self.imageURLs.first
+    }
```

Comment:
@lburdock @guido732 icon is the first image and background is the second. That was completely arbitrary and I'm happy to change it, as long as we do the same :)

Reply:
personally I'd do it the other way around just to keep consistency with single-image templates, where it's most likely they'll always be the background or main image, the second or rest being less-important ones.

VERY nit-picky

Reply:
Makes sense actually 👍🏻

Reply:
Changed.